File: core/internet/evidence_consensus_engine.py
from __future__ import annotations

from typing import Any, Dict, List

from core.internet.source_corroboration_engine import corroborate_sources
from core.internet.semantic_consensus_engine import measure_semantic_consensus


def build_evidence_consensus(sources: List[Dict[str, Any]], claims: List[Dict[str, Any]]) -> Dict[str, Any]:
    corr = corroborate_sources(sources)
    consensus = measure_semantic_consensus(claims)
    strength = round(min(1.0, corr.get("corroboration_count", 0) * 0.2 + consensus.get("consensus", 0) * 0.8), 3)
    return {
        "corroboration": corr,
        "consensus": consensus,
        "strength": strength,
        "deterministic_inputs": [f"strength={strength}"],
    }

File: src/internet/evidenceConsensusEngine.ts
/**
 * Converted from Python: core/internet/evidence_consensus_engine.py
 * @generated — WebWeaveX python→javascript library port
 */

import * as py from "../runtime/pyCompat.js";
import { corroborateSources } from "./sourceCorroborationEngine.js";
import { measureSemanticConsensus } from "./semanticConsensusEngine.js";

export function buildEvidenceConsensus(sources: any, claims: any): any {
  var corr: any = corroborateSources(sources);
  var consensus: any = measureSemanticConsensus(claims);
  var strength: any = py.round(py.min([py.F(1.0), py.add(py.mul(py.get(corr, "corroboration_count", 0), py.F(0.2)), py.mul(py.get(consensus, "consensus", 0), py.F(0.8)))]), 3);
  return {"corroboration": corr, "consensus": consensus, "strength": strength, "deterministic_inputs": [`strength=${py.floatStr(strength)}`]};
}
export { corroborateSources, measureSemanticConsensus };
